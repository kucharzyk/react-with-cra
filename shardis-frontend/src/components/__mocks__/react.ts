import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
