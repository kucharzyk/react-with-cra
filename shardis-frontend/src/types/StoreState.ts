import { CounterState } from '../reducers/counter';

export default interface StoreState {
  counter: CounterState;
}
