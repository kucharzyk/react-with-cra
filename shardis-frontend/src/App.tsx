import * as React from 'react';
import './App.css';
import ServerTime from './components/ServerTime';
import Header from './components/Header';

class App extends React.Component {
  render() {
    return (
      <div className="App">
        <Header/>
        <p className="App-intro">
          To get started, edit <code>src/App.tsx</code> and save to reload.
        </p>
        <ServerTime/>
      </div>
    );
  }
}

export default App;
