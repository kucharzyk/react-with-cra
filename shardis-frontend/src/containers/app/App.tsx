import * as React from 'react';
import { StatelessComponent } from 'react';
import { StyleRulesCallback, Theme } from 'material-ui/styles';
import withStyles from 'material-ui/styles/withStyles';
import { WithStyles } from 'material-ui';
import AppHeader from '../../components/app-header/AppHeader';
import AppRoutes from '../app-routes/AppRoutes';
import AppFooter from '../../components/app-footer/AppFooter';

const styles: StyleRulesCallback<'root'> = (theme: Theme) => ({
  root: {
    padding: theme.spacing.unit
  },
});

const App: StatelessComponent<{} & WithStyles<'root'>> = (props) => (
  <div className={props.classes.root}>
    <AppHeader/>
    <AppRoutes/>
    <AppFooter/>
  </div>
);

export default withStyles(styles)<{}>(App);
