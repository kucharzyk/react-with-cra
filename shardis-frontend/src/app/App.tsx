import * as React from 'react';
import './App.css';
import createBrowserHistory from 'history/createBrowserHistory';
import { Route, Router, Switch } from 'react-router-dom';
import MainLayout from '../layout/main-layout/MainLayout';
import createMuiTheme from 'material-ui/styles/createMuiTheme';
import purple from 'material-ui/colors/purple';
import green from 'material-ui/colors/green';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Reboot from 'material-ui/Reboot';

const theme = createMuiTheme({
  palette: {
    primary: purple,
    secondary: green,
  },
});

class App extends React.Component {
  render() {
    return (
      <MuiThemeProvider theme={theme}>
        <Reboot/>
        <Router history={createBrowserHistory()}>
          <Switch>
            <Route path={'/'} component={MainLayout}/>
          </Switch>
        </Router>
      </MuiThemeProvider>
    );
  }
}

export default App;
