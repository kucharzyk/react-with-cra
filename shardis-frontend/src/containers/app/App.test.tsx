import * as React from 'react';
import { StatelessComponent } from 'react';
import App from './App';
import createShallow from 'material-ui/test-utils/createShallow';
import toJson from 'enzyme-to-json';
import StoreState from '../../types/StoreState';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';

describe('App', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {

    const mockStore = configureStore<Partial<StoreState>>();
    const store = mockStore({});
    WrappedComponent = () => (
      <Provider store={store}>
        <StaticRouter location={'/'} context={{}}>
          <App/>
        </StaticRouter>
      </Provider>
    );
  });

  it('renders snapshot', () => {
    const shallowComponent = createShallow({untilSelector: 'App'})(<WrappedComponent/>);
    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

});
