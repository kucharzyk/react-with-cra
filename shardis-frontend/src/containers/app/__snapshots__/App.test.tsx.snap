// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<div
  className="App-root-1"
>
  <withRouter(WithStyles(AppHeader)) />
  <WithStyles(AppRoutes) />
  <WithStyles(AppFooter) />
</div>
`;
